internal/daemon/controller/handlers/targets/target_service.go

```diff
@@ -1585,6 +1630,20 @@ func validateUpdateRequest(req *pbs.UpdateTargetRequest) error {
 				badFields[globals.IngressWorkerFilterField] = err.Error()
 			}
 		}
+		if address := req.GetItem().GetAddress(); address != nil {
+			if len(address.GetValue()) < static.MinHostAddressLength ||
+				len(address.GetValue()) > static.MaxHostAddressLength {
+				badFields[globals.AddressField] = fmt.Sprintf("Address length must be between %d and %d characters.", static.MinHostAddressLength, static.MaxHostAddressLength)
+			}
+			_, _, err := net.SplitHostPort(address.GetValue())
+			switch {
+			case err == nil:
+				badFields[globals.AddressField] = "Address does not support a port."
+			case strings.Contains(err.Error(), "missing port in address"):
```

Comment:
Addressed [28fb195](https://github.com/hashicorp/boundary/pull/2613/commits/28fb195b4b4be8ca8e801ecfb432601a883782d2)

Reply:
Not sure if unifying here makes sense because the hosts handlers also have this same code - We decided at the time not to extract because we couldn't find a sensible place to put the unified code.

For me it makes sense to either unify all or not at all - But unifying the entire thing means finding a suitable place for the code

Reply:
I agree with Hugo, I'm not sure where we could put the code in the current structure. We might need to add a new package called address or network. But then, this would require moving everything network address related in the repo to this location, in order to maintain consistency. 
It might turn out to be more work and maybe not in the scope of this feature.